### DISCUSSION TITLE
`turbo` model release

### DISCUSSION THREAD

**Discussion by jongwook** [openai/whisper#2363] [Announcements]
We’re releasing a new Whisper model named `large-v3-turbo`, or `turbo` for short. It is an optimized version of [Whisper `large-v3`](https://github.com/openai/whisper/discussions/1762) and has only 4 decoder layers—just like the `tiny` model—down from the 32 in the `large` series. 

This work is inspired by [Distil-Whisper](https://arxiv.org/abs/2311.00430)<sup>[1]</sup>, where the authors observed that using a smaller decoder can greatly improve transcription speed while causing minimal degradation in accuracy. Unlike Distil-Whisper, which used distillation to train a smaller model, Whisper turbo was fine-tuned for two more epochs over the same amount of multilingual transcription data used for training `large-v3`, i.e. excluding translation data, on which we don’t expect turbo to perform well.

Across languages, the `turbo` model performs similarly to `large-v2`, though it shows larger degradation on some languages like Thai and Cantonese. Whisper `turbo` performs better on FLEURS, which consists of cleaner recordings than Common Voice. The figure below shows the `turbo` model’s performance on the subset of languages in the Common Voice 15 and FLEURS datasets where `large-v3` scored a 20% error rate or lower.

![language-breakdown](https://github.com/user-attachments/assets/d785c8ca-ffa1-47a3-bdad-9dcb4265f1c0)

Combined with a recent patch (#2359) to use `F.scaled_dot_product_attention` when available, the ASR speed of `turbo` is faster than what `tiny` used to be:

![whisper-speed](https://github.com/user-attachments/assets/f85b048a-0ecf-4da1-942a-fd0309d11ccc)

This puts `turbo` at the "best of both worlds" when comparing speed and accuracy:

![performance-vs-speed](https://github.com/user-attachments/assets/336c5257-76e2-4420-8354-b5b27292d853)

---

You can update the Python package to version `20240930` or later to use the new model:

```bash
pip install -U openai-whisper
```

With this upgrade, the `whisper` CLI command now defaults to the `turbo` model:

```bash
whisper audio.wav  # will use `--model turbo` by default
```

---

### References
> [1] Gandhi, S., von Platen, P., & Rush, A. M. (2023). Distil-whisper: Robust knowledge distillation via large-scale pseudo labelling. *arXiv preprint arXiv:2311.00430.*

**Comment by FurkanGozukara** [comment]
it is amazing but i really want to to see lower WER rather than speed up :( I hope it arrives asap

**Comment by AmgadHasan** [comment]
Thanks you @jongwook for releasing this.

For finetuning, what learning rate would you recommend? We've been using 1e-5 with large-v2 and large-v3 and it's been working well. Should we change it for turbo?

**Reply by FurkanGozukara** [reply]
> Thanks you @jongwook for releasing this.
> 
> For finetuning, what learning rate would you recommend? We've been using 1e-5 with large-v2 and large-v3 and it's been working well. Should we change it for turbo?

hi any good tutorial for fine tuning easy tools?

i would like to fine tune on my fully accurate subtitles having videos

https://www.youtube.com/SECourses

**Reply by LaurinmyReha** [reply]
maybe give this one a shot. Should be a bit more accurate and deliver better timestamps for the subtitles: 

https://github.com/nyrahealth/CrisperWhisper

**Reply by FurkanGozukara** [reply]
@LaurinmyReha thanks it looks great

 why it is not merged here? i cant make sense of it

**Reply by jongwook** [reply]
Hi, i used a linear LR decay starting from 5e-5 for this model, so 1e-5 sounds like a good start, although it will depend on the amount and diversity of the dataset. I'd recommend doing a LR sweep and checking validation metrics, as usual.

**Reply by FurkanGozukara** [reply]
@jongwook what do we use how do we train? example code repo?

**Reply by jongwook** [reply]
HF has an excellent article on this: https://huggingface.co/blog/fine-tune-whisper and also see #64

**Reply by FurkanGozukara** [reply]
@jongwook thank you so much. will this notebook work for newest turbo model? https://colab.research.google.com/github/sanchit-gandhi/notebooks/blob/main/fine_tune_whisper.ipynb

**Reply by asr-lord** [reply]
> @jongwook thank you so much. will this notebook work for newest turbo model? https://colab.research.google.com/github/sanchit-gandhi/notebooks/blob/main/fine_tune_whisper.ipynb

Yes, you should change `openai/whisper-small` to `openai/whisper-large-v3-turbo` in the notebook.

**Reply by gslin1224** [reply]
Hi everyone,

I have a question about Whisper's translate function and fine-tuning.

If I fine-tune the Whisper large model using a dataset that includes mixed Chinese and English audio, along with bilingual transcripts (Chinese and English), will this fine-tuning process affect the model's translate functionality? Specifically, after fine-tuning, will I still be able to use Whisper's built-in translation feature to translate Chinese audio into English?

Thanks in advance for any insights!

**Comment by flishwang** [comment]
Does the turbo model still support translate task? I tried it but it seems outputing the raw languages instead of English. Does any one meet this issue?

**Reply by Jiltseb** [reply]
It's not finetuned for `translation` task, so it won't perform well as @jongwook already written.

**Comment by Vaibhavs10** [comment]
Hi all, I'm VB from the Open Source audio team at Hugging Face. Thanks a ton @jongwook and team for `Whisper-large-v3-turbo` - It's amazing to see the results.

For reference, the `transformers` compatible weights are here: https://huggingface.co/openai/whisper-large-v3-turbo

You can reference them via a simple snippet:

```python
import torch
from transformers import pipeline

model_id = "openai/whisper-large-v3-turbo"

pipe = pipeline(
    "automatic-speech-recognition",
    model=model_id, 
    torch_dtype=torch.float16,
    device="cuda" # replace with `mps` for Mac
)

result = pipe("<file_name.mp3>") # pass `return_timestamps=True` for timestamps
print(result["text"])
```

You can also fine-tune it for your own use-case as well: https://huggingface.co/blog/fine-tune-whisper

Looking forward to the next updates to the series 🤗

**Reply by FurkanGozukara** [reply]
@Vaibhavs10 any easy tool guide tutorial to fine tune? i would like to fine tune on my video which as fully accurate subtitles - written by me : https://www.youtube.com/SECourses

**Reply by AmgadHasan** [reply]
@Vaibhavs10 
@jongwook 
What is the native precision of the model? The HF version is fp16. Shouldn't it be bf16?

**Comment by Herz3h** [comment]
Amazing work !

Thank you. Is there any ongoing work regarding the hallucination of whisper ? This for us seems to be the biggest downside of using it. Sometimes it also gets stuck in a loop and keeps repeating stuff (happens more often in v3, therefore we use v2 which seems better in French language at least)

**Reply by gongouveia** [reply]
Hello @Herz3h
Check about enabling VAD, supressing silent instances should improve your hallucinations. 
You can activate whisper's VAD or use a more robust pre processing with it (e.g. Silero).

**Reply by jhj0517** [reply]
You can use [Silero VAD](https://github.com/snakers4/silero-vad) to detect voice parts, or [UVR](https://github.com/Anjok07/ultimatevocalremovergui) to remove other noise (such as music) from the audio. 

In [Whisper-WebUI](https://github.com/jhj0517/Whisper-WebUI) such pre-processing pipeline is implemented for lower WER. 
Of course, since it uses submodels in the pipeline, it slows down the whole process when you use it.

**Reply by gongouveia** [reply]
@Herz3h @jhj0517 Beware that using noise removal (such as Demucs in UVR) although enhances hearing quality, it adds impurities and artifacts to the frequency domain, significantly decreasing translation quality (higher WER/CER).
Whisper is more than enough capable of translating with some background noise, no need of noise removal in most cases.

**Reply by jhj0517** [reply]
I used [MDX](https://github.com/ZFTurbo/MVSEP-MDX23-music-separation-model) models for noise removal and in my test it gave me better results in most cases.

I haven't tested with Demucs, so I don't know if it could negatively affect the result.

But yes, there's still a possibility that submodels can also cause hallucinations, so you might want to compare results with submodels / without submodels.

I prefer to use them when the audio has such noise, because in my experience it gives me better results in most cases.

**Reply by xmoiduts** [reply]
I recommend cutting the beginning of the media file and making the audio appear at the start of the media, this suppresses hallucinations related to the topic but should not exist, and, in word-precise mode, shorten the 'words' so that they are no longer multi-sentences per word.

**Comment by ghost** [comment]
Any chance for availability of this model over the offical OpenAI API anytime soon?

**Comment by simpthy** [comment]


**Reply by ryanheise** [reply]
@simpthy 
According to the v3 release, some languages without spaces to separate words, such as Japanese and Mandarin, were measured in CER rather than WER. I think Korean was also measured in CER (Korean does have spaces, although it doesn't use spaces to separate everything, and it's comparable in syntax to Japanese.)

**Reply by jongwook** [reply]
Yes to what Ryan said; the languages in *italic* used CER. I've switched to use CER for Korean after realizing that WER is often too harsh because the use of spaces is often arbitrary in many Korean words. CER also fits better with the agglutinative nature of the language.

I'm not sure what exactly contributed to the better accuracy in the Taiwan subset and haven't done any deeper analysis; the reason could be anything from - the subset had more sentences that are easier to recognize, the recording quality was better in general, or the model actually better understands the way people speak Mandarin in Taiwan.

**Reply by ryanheise** [reply]
This may be helpful: https://huggingface.co/learn/audio-course/en/chapter5/evaluation#which-metric-should-i-use

> In general, the WER is used far more than the CER for assessing speech systems. This is because the WER requires systems to have greater understanding of the context of the predictions. In our example, “sit” is in the wrong tense. A system that understands the relationship between the verb and tense of the sentence would have predicted the correct verb tense of “sat”. We want to encourage this level of understanding from our speech systems. So although the WER is less forgiving than the CER, it’s also more conducive to the kinds of intelligible systems we want to develop. Therefore, we typically use the WER and would encourage you to as well! However, there are circumstances where it is not possible to use the WER. Certain languages, such as Mandarin and Japanese, have no notion of ‘words’, and so the WER is meaningless. Here, we revert to using the CER.

**Comment by ushnah** [comment]
where can i find WER for urdu language (turbo model)?

**Reply by jongwook** [reply]
Turbo scored 26.5% (CV15) and 21.9% (FLEURS) on Urdu

**Reply by MedAymenF** [reply]
@jongwook What about Kabyle (kab)?

**Comment by zx3777** [comment]
Thanks, the biggest drawback of Whisper right now is that the timeline is inaccurate. Subtitles generated by the large series appear much earlier than they should. Tested in korean.

**Comment by ioudove** [comment]
Could you please find a way to solve the old issues of hallucinations and incorrect timestamps in Whisper?

**Reply by LaurinmyReha** [reply]
maybe give this one a shot. Should be a bit more accurate and deliver better timestamps, atleast for english:

https://github.com/nyrahealth/CrisperWhisper

**Comment by oswaldoludwig** [comment]
To speed up decoding without hurting performance too much, we should have a fat, shallow decoder, to take advantage of parallel processing. Fewer layers in the decoder, as they did, but more attention heads per layer, this could be computed in parallel, but the layers cannot (i.e. we can only compute a layer after we have its input, which is the output of the previous one). By keeping the encoder deep, you ensure that the input is thoroughly processed and rich in features. Then, using a fat, shallow decoder can speed up the auto-regressive decoding process, which calls the decoder block many times.

**Comment by Jeong-Haneul** [comment]
I’ve noticed an issue with the no_speech_prob variable when using the Turbo model. While it works correctly with the large model, it doesn’t seem to function as expected in the Turbo model. Please look into this.

Thank you!
'no_speech_prob': 1.903758928278876e-09
in turbo model.

**Comment by FurkanGozukara** [comment]
@jongwook any idea to fix this bad broken Turbo model?

on my newest video whisper turbo performed really really bad - it was so broken that unfixable manually - i had to use large-v3 and fix it

here turbo model transcription : 
[turbo model.txt](https://github.com/user-attachments/files/17242734/turbo.model.txt)

### And here fully manually fixed subtitles video (you can download and test) : https://youtu.be/URnOHbmuKWs

here manually fixed subtitles : 
[manually fixed.txt](https://github.com/user-attachments/files/17242736/manually.fixed.txt)

this is why we need a better WER model not a faster one  - it is unusable without huge effort 

compare both and you won't believe difference it is just huge sadly

here example 2 case

### below one turbo

![image](https://github.com/user-attachments/assets/50589fb0-8f15-4c6b-ac3f-6b409523c918)


###  below one manually fixed
![image](https://github.com/user-attachments/assets/ef84f2d4-b8d1-4781-8e40-2f0969c75554)

**Reply by AeneasZhu** [reply]
I also encountered this problem before. I tried fixing it by adding `--word_timestamps True` parameter, which could help improve the timestamps.  Here is my result in the same part as yours:
![image](https://github.com/user-attachments/assets/ab1b0dcd-032e-4025-ae08-5e5475957ba8)
However, I encountered another problem: whisper lost the punctuation:
![image](https://github.com/user-attachments/assets/7a8ac5ac-8b7c-4dca-8a90-42a6ce23ea0d)
And I tried fixing it by adding `initial_prompt`. Here is my final solution: `whisper audio.mp3 --initial_prompt "Include all punctuation, new paragraph, including (brackets), new paragraph, in the transcription" --word_timestamps True`.

The `--initial_promt` forced punctuation mode mannually and `--word_timetamps True` could refine the timestamps level. Hope this could help.

**Reply by ryanheise** [reply]
> Here is my final solution: `whisper audio.mp3 --initial_prompt "Include all punctuation, new paragraph, including (brackets), new paragraph, in the transcription" --word_timestamps True`.

It's worth noting that the reason this may help is not because you are instructing Whisper to include all punctuation (it is not like a ChatGPT prompt), but simply because your prompt contains examples of punctuation. All that happens when you supply the initial prompt is that you are telling whisper the transcript of what came just before the current point in the audio. So here, Whisper will think that if we rewound the audio 30 seconds, the previous part of audio would have been transcribed literally as "Include all punctuation, new paragraph, including (brackets), new paragraph, in the transcription". If that prior transcript contains punctuation, it makes it more likely that Whisper will include punctuation in the transcript of the next chunk of audio that follows (Perhaps you can design a better prompt with that understanding). This also works for vocabulary: if you incorporate certain vocabulary into the prompt, Whisper will be more likely to use the spelling you have provided in the next part of the transcript.

**Reply by FurkanGozukara** [reply]
thank you for replies @ryanheise and @AeneasZhu . i was already using this : 

--initial_prompt ""Welcome to our Youtube channel.""

but it still loses punctuating in many cases which i hate

I haven't tried --word_timestamps True with turbo gonna try

**Reply by ryanheise** [reply]
Your prompt doesn't contain much punctuation (only one full stop, and no commas), so you might consider making your prompt longer so that it has enough punctuation, to give it a feel for what you want. It is still possible for the initial prompt to eventually become a distant/forgotten memory in long audio files, so if you're experiencing that behaviour, you might want to test this PR: #2343

**Reply by FurkanGozukara** [reply]
@ryanheise do you have good prompt?

**Reply by ryanheise** [reply]
There isn't a general purpose prompt I can recommend, so I think you'll just need to experiment to find what works well with your specific audio domain (e.g. Try short vs long prompts, multiple sentences vs one sentence, vary the amount of punctuation, and so on), but also I suggest testing #2343 in case you find that the transcription starts out well and then degrades later in the audio.

**Comment by gslin1224** [comment]
Hey guys, if I wanna use
 ```
Adding --task translate will translate the speech into English:
whisper japanese.wav --language Japanese --task translate
```
in the Python Usage,what library should I call? 

thanks a lot!

I try It and found that solution to use python usage,

below is the code for Python usage
```
import whisper

model = whisper.load_model("large", download_root="/home/Lin/whisper")

result = model.transcribe("/home/common_voice_zh-CN_21347786.mp3", task="translate")

print(result["text"])

```

But current turbo seem doesn't support that translate Function

**Reply by Sekiraw** [reply]
It's not finetuned for translation task, so it won't perform well as @jongwook already written.

**Reply by gslin1224** [reply]
> It's not finetuned for translation task, so it won't perform well as @jongwook already written.

So now only turbo not support the translate Function?

**Reply by asr-lord** [reply]
As they said, turbo not perform well in translation: _"Whisper turbo was fine-tuned for two more epochs over the same amount of multilingual transcription data used for training large-v3, i.e. excluding translation data, on which we don’t expect turbo to perform well."_

**Reply by gslin1224** [reply]
> As they said, turbo not perform well in translation: _"Whisper turbo was fine-tuned for two more epochs over the same amount of multilingual transcription data used for training large-v3, i.e. excluding translation data, on which we don’t expect turbo to perform well."_

ok, thanks a lot

**Reply by meekmeekmeek** [reply]
Is the limitation with turbo not performing well with translation something inherent with the fewer-layers approach? For a lot of people the main usecase for whisper is generating subtitles for non-english languages, so losing out on the very significant speed-up from large-v3 to turbo is a shame.

Also since turbo seems to be the default model no, you really should consider displaying some kind of warning when turbo is used with the "translate" option, rather than going through the whole transcription without translating. I know an experienced user will notice early (if they're watching the output) that only transcription is happening, but I'm willing to be more than one user has waited on a very long transcription process and been left wondering why nothing was translated since the CLI reference doesn't mention turbo not supporting the translate option.

**Reply by ryanheise** [reply]
> Also since turbo seems to be the default model no, you really should consider displaying some kind of warning when turbo is used with the "translate" option, rather than going through the whole transcription without translating.

I had a similar thought. The warning is a good idea, and it may also be helpful to have a different default model for `translate` so that translation works out of the box.

**Comment by x86Gr** [comment]
I've been using the turbo model and I'm getting moderately better quality, in proper names and acronyms compared to V3. With a moderately noisy audio file i get things like this:

Large V3: We're coming back from pruxelles were we held a CDA to fix the mess of the other day. we did it but we're late like jankees. you know what I mean
Turbo: We're coming back from Bruxelles were we held a CDA to fix the mess of the other day. We did it but we're late like...like jankees! You know what I mean?

Turbo compared to V3 behaves more like a Large V2 with clean audio.
Settings are no prompt / condition on previous text False / beam size 5 / best of 5 / patience 1.0
(we're still testing it so the sampled audios are not many)

**Comment by dgoryeo** [comment]
When the model is downloaded at the first useage the size of the zip file is roughly 1.5GB, while I expected `large-v3-turbo` to be 6GB. Is that purely due to the compression, or might have something gone wrong in my environment?

**Reply by gordon0414** [reply]
just as @jongwook mentioned in his first paragraph,
```
It is an optimized version of https://github.com/openai/whisper/discussions/1762 and has only 4 decoder layers—just like the tiny model—down from the 32 in the large series.
```

the decoder is 1/8 wise, and when we think of the fact that the decoder part would be originally bigger in size because of the cross-attention KQV, we can understand the decrease of the size

**Comment by andyhoeung** [comment]
Personally, when transcribing Japanese audio, I still see some random unrelated English words being inserted inbetween Japanese characters. Seems this issue still persists since v3, and v2 doesn't have this problem. Shame, because it's really fast and outside of that issue, it's really good overall.

**Reply by misutoneko** [reply]
That's funny, I think I got it the other way around at some point (japanese characters for english audio).
Have you tried with --hallucination_silence_threshold? In my testing that got rid of most(?) of the hallucinations.
I've only used english audio samples though.

**Reply by andyhoeung** [reply]
I'll give that a try and report back.

**Reply by andyhoeung** [reply]
I started with a value of 2 up until 8, didn't seem to fix much.

**Reply by misutoneko** [reply]
Thank you for testing. My samples are very very short (under 30s mostly) so that may also have something to do with it.
I used smaller values such as 0.3 or so...but yeah, gotta keep experimenting I guess :D

**Comment by zxl777** [comment]
When inputting an 11-hour MP3 file, the error rate increases significantly, and several lines were missed in the transcription of the first 20 minutes. 

`time whisper '11hours.mp3' --model turbo  --output_format json --word_timestamps True
`

However, when I extracted the first 20 minutes of audio and transcribed it again, everything worked fine.

`ffmpeg -i 11hours.mp3 -t 1200 -acodec copy 20_minutes.mp3`

`time whisper 20_minutes.mp3 --model turbo  --output_format json --word_timestamps True`

**Reply by xmoiduts** [reply]
Will it help that clipping the start to the first word? Like -ss <just before the first spoken word> [-t some seconds]?

**Reply by dgoryeo** [reply]
Have you considered this approach perhaps: [WhisperHallu](https://github.com/EtienneAb3d/WhisperHallu)

**Reply by zxl777** [reply]
After continuous attempts, I found that if I want to generate a JSON file with timestamps, adding an initial_prompt brings back the missing sentence.

`initial_prompt='They internalize the cultural message of “It’s your fault! You should exercise more, but you aren’t doing it. Shame on you!” I am here to say: It isn’t your fault.'
`


Additionally, if I only generate plain text without timestamps, the result is also correct.
`time whisper '11hours.mp3' --model turbo --output_format txt`

**Reply by DoS007** [reply]
@zxl777 Does this happen with the original whisper v3 turbo, or do you use `deepdml/faster-whisper-large-v3-turbo-ct2` ? If you use the last one, try `mobiuslabsgmbh/faster-whisper-large-v3-turbo` instead (see [here and post above there](https://github.com/SYSTRAN/faster-whisper/issues/1030#issuecomment-2440087219)).

**Reply by zxl777** [reply]
@DoS007 
Yes, I am now using the faster-whisper-large-v3-turbo-ct2. 
Thank you for your suggestion; I will update the faster-whisper to use mobiuslabsgmbh/faster-whisper-large-v3-turbo.

**Comment by DoS007** [comment]
Hmm, what I don't really understand is that there seem to be more problems with v3 regarding hallucinations and perhaps missing sentences than with v2. Even OpenAI uses V2 and *not* V3 for their api (see [here](https://platform.openai.com/docs/guides/speech-to-text/introduction)). So they might be aware of this as well.

So why not use large-v2 as a basis for turbo instead of large-v3? I mean, robustness is more important than a slightly lower benchmark number?

**Comment by Francoyy** [comment]
I don't understand, the Turbo behaves so bad for Mandarin in terms of chopping the text in multiple sections. And it seems like it's using "1" as a kind of separator, but i don't see why it would do so. Nobody got anything like that in the last 3 months?

large-v2:
```
[02:19.020 --> 02:21.400] 我的腦子可能晚上睡不著
[02:21.400 --> 02:23.300] 所以我覺得那怎麼辦
[02:23.300 --> 02:26.320] 我白天,早上白天不能剪片
[02:26.320 --> 02:28.960] 晚上也不太可以剪片
[02:28.960 --> 02:31.600] 那我到底什麼時候可以剪片
[02:31.600 --> 02:36.540] 所以這兩年發影片的速度可能大家有發現變慢
[02:36.540 --> 02:42.640] 我現在平均好像是兩個或三個禮拜才會出一個影片
```

turbo:
```
[02:09.380 --> 02:36.500] １可能晚上睡不著１所以１我覺得那怎麼辦１我白天１早上１白天１不能剪片１晚上１也不太可以剪片１那我到底什麼時候可以剪片１所以１這兩年１發影片１的速度１可能大家有發現１變慢�我現在平均�好像１
```

(To add that large-v3 has always been bad for me and my other friends case for Mandarin. It doesn't seem to tolerate silences through the video. When there's a silence of a few seconds, say 10 seconds, it often completely crashes and stops transcribing. But those blanks are perfectly normal for video editing, when music is added later on).
